src/AutoRest.CSharp/Common/Input/InputTypes/InputModelType.cs

```diff
@@ -6,9 +6,14 @@
 
 namespace AutoRest.CSharp.Common.Input
 {
-    internal record InputModelType(string Name, string? Namespace, string? Accessibility, string? Deprecated, string? Description, InputModelTypeUsage Usage, IReadOnlyList<InputModelProperty> Properties, InputModelType? BaseModel, IReadOnlyList<InputModelType> DerivedModels, string? DiscriminatorValue, string? DiscriminatorPropertyName, bool IsNullable)
+    internal record InputModelType(string Name, string? Namespace, string? Accessibility, string? Deprecated, string? Description, InputModelTypeUsage Usage, IReadOnlyList<InputModelProperty> Properties, IReadOnlyList<InputModelType> DerivedModels, string? DiscriminatorValue, string? DiscriminatorPropertyName, bool IsNullable)
         : InputType(Name, IsNullable)
     {
+        public InputModelType(string name, string? ns, string? accessibility, string? deprecated, string? description, InputModelTypeUsage usage, IReadOnlyList<InputModelProperty> properties, InputModelType? baseModel, IReadOnlyList<InputModelType> derivedModels, string? discriminatorValue, string? discriminatorPropertyName, bool isNullable)
+            : this(name, ns, accessibility, deprecated, description, usage, properties, derivedModels, discriminatorValue, discriminatorPropertyName, isNullable)
+        {
+            BaseModel = baseModel;
+        }
```

Comment:
these changes are unacceptable.
According to our discussion yesterday, these classes (this one and the corresponding converter/deserializer) do not need to be changed, it is already possible to handle the case we are currently having.
The only issue is in the emitter.
If we would like to introduce a temporary workaround, I prefer the previous way which is simpler and straightforward to revert.

Reply:
And I have a try on my machine, by checking out to your branch, revert all the changes in CodeModelConverter.cs, InputModelType.cs, and TypeSpecInputModelTypeConverter.cs, then regenerate FirstTest-TypeSpec, it works fine and generated code does not have a change.
Is there anything I am missing that we really require these changes?

Reply:
I put the changes in this PR: https://github.com/Azure/autorest.csharp/pull/4038